### PR TITLE
Add distinct function to MongoHook in apache-airflow-providers-mongo

### PR DESCRIPTION
**Pull request by octaviolage**
---

This PR adds a `distinct()` function to the `MongoHook` in `apache-airflow-providers-mongo`, which reffers [pymongo.collection.distinct](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.distinct). This function allows users to return a list of distinct values from a MongoDB collection. I think the distinct function fits well with the other collection functions in the hook.

Before:
```py
hook = MongoHook(conn_id='my_conn_id')
coll = hook.get_collection(
                mongo_db='test',
                mongo_collection='test',
            )
distinct_keys = coll.distinct('name')
```
After:
```py
hook = MongoHook(conn_id='my_conn_id')
distinct_keys= hook.get_collection(
                mongo_db='test',
                mongo_collection='test',
                key='name'
            )
```

### CODE REVIEW

File: airflow/providers/mongo/hooks/mongo.py [hunk 1]
```diff
@@ -366,3 +366,22 @@ def delete_many(
         collection = self.get_collection(mongo_collection, mongo_db=mongo_db)
 
         return collection.delete_many(filter_doc, **kwargs)
+
+    def distinct(
+        self, mongo_collection: str, key: str, filter_doc: dict | None, mongo_db: str | None = None, **kwargs
```
Comment (Lee-W) [inline airflow/providers/mongo/hooks/mongo.py]: Do we want to name `key` as `distinct_key` instead? Or is it intuitive enough for users to understand?

File: airflow/providers/mongo/hooks/mongo.py [hunk 2]
```diff
@@ -366,3 +366,22 @@ def delete_many(
...
+    ) -> list[Any]:
+        """
+        Returns a list of distinct values for the given key across a collection.
+
+        https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html#pymongo.collection.Collection.distinct
+
+        :param mongo_collection: The name of the collection to perform distinct on.
+        :param key: The field to return distinct values from.
+        :param filter_doc: A query that matches the documents get distinct values from.
+            Optional. Defaults to {}.
```
Comment (Lee-W) [inline airflow/providers/mongo/hooks/mongo.py]: This seems to be inaccurate. There's no default to the parameter. Should we add it or change the docstring?

**Review by vincbeck — APPROVED** [review]

**Comment by vincbeck** [comment]
There are some test failures. Please run [static checks](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst) and also check the unit test you wrote, it is failing.

File: airflow/providers/mongo/hooks/mongo.py [hunk 3]
```diff
@@ -366,3 +366,22 @@ def delete_many(
         collection = self.get_collection(mongo_collection, mongo_db=mongo_db)
 
         return collection.delete_many(filter_doc, **kwargs)
+
+    def distinct(
+        self, mongo_collection: str, distinct_key: str, filter_doc: dict | None = None, mongo_db: str | None = None, **kwargs
```
Comment (Taragolis) [inline airflow/providers/mongo/hooks/mongo.py]:
```suggestion
        self,
        mongo_collection: str,
        distinct_key: str,
        filter_doc: dict | None = None,
        mongo_db: str | None = None,
        **kwargs,
```

**Review by vincbeck — APPROVED** [review]

**Pull request merged**